fedora/client/openidbaseclient.py

```diff
@@ -223,26 +233,105 @@ def send_request(self, method, auth=False, verb='POST', **kwargs):
         except KeyError:
             raise Exception('Unknown HTTP verb')
 
-        try:
-            output = func(method, **kwargs)
-        except LoginRequiredError:
-            raise AuthError()
+        if retries is None:
+            retries = self.retries
 
-        try:
-            data = output.json
-            # Compatibility with newer python-requests
-            if callable(data):
-                data = data()
-        except ValueError as e:
-            # The response wasn't JSON data
-            raise ServerError(
-                method, output.status_code, 'Error returned from'
-                ' json module while processing %(url)s: %(err)s\n%(output)s' %
-                {
-                    'url': to_bytes(method),
-                    'err': to_bytes(e),
-                    'output': to_bytes(output.text),
-                })
+        if timeout is None:
+            timeout = self.timeout
+
+        retry_sleep = retry_sleep or self.retry_sleep
```

Comment:
There is one slight problem with this, `send_request(retry_sleep=0)` won't work correctly. You have to compare with `None` directly.